wallet/udb/vsp.go

```diff
@@ -35,23 +48,51 @@ func GetVSPTicket(dbtx walletdb.ReadTx, tickethash chainhash.Hash) (*VSPTicket,
 	return ticket, nil
 }
 
+// GetVSPTicketsByFeeStatus gets all vsp tickets which have
+// FeeTxStatus == feeStatus.
+func GetVSPTicketsByFeeStatus(dbtx walletdb.ReadTx, feeStatus int) (map[chainhash.Hash]*VSPTicket, error) {
+	bucket := dbtx.ReadBucket(vspBucketKey)
+	tickets := make(map[chainhash.Hash]*VSPTicket)
+	err := bucket.ForEach(func(k, v []byte) error {
+		ticket := deserializeVSPTicket(v)
+		if int(ticket.FeeTxStatus) == feeStatus {
+			var hash chainhash.Hash
+			hash.SetBytes(k)
+			tickets[hash] = ticket
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tickets, nil
+}
+
 // deserializeUserTicket deserializes the passed serialized user
 // ticket information.
 func deserializeVSPTicket(serializedTicket []byte) *VSPTicket {
-
+	// ticket stores hash size and an uint32 representing the fee processment
+	// status.
+	curPos := 0
 	vspTicket := &VSPTicket{}
-	copy(vspTicket.FeeHash[:], serializedTicket[:])
+	copy(vspTicket.FeeHash[:], serializedTicket[curPos:hashSize])
+	curPos += hashSize
+	vspTicket.FeeTxStatus = byteOrder.Uint32(serializedTicket[curPos : curPos+4])
```

Comment:
this will totally break with any vsp tickets recorded in the db with the previous serialization format.

Reply:
resolved by a db upgrade